### ISSUE TITLE
Migrate to Yarn v3

### ISSUE THREAD

**Issue by andrewpeters9** [ankitects/anki#2430]
**Why:**
* Along with a few other benefits, Yarn 3 ships with the ability to patch packages natively
* Anki currently uses `patch-package` which could be replaced by `yarn@3.x`
* Setup process would be slightly simplified by moving the patches to yarn's native patching functionality

I'm happy to open up an MR for this myself - if this issue is approved

**Comment by dae** [comment]
My only concern would be possible regressions, but provided your PR can be easily rolled back if they do crop up (until the issues are addressed), and if you're willing to do the work, I can't think of a good reason to say no.

You mention faster installs - are you referring to PnP? The build process currently copies certain files out of node_modules to redistribute them, so we'd need to make sure that still works somehow.

In the medium term, I hope that we can actually move to deno or bun. The main blocker there is the Svelte tooling as far as I'm aware - if it means we lose IDE support, then that's going to be hard to justify.

**Comment by andrewpeters9** [comment]
> You mention faster installs - are you referring to PnP? The build process currently copies certain files out of node_modules to redistribute them, so we'd need to make sure that still works somehow.

Correct, but from experience, this typically isn't a problem: as you node environment will still have access to those `node_modules` files, it's just that they're now sym-linked to other places in your local setup.

If Deno/Bun migration is a medium-term goal then maybe this issue should be ignored.

> FYI, as a bit of a tangent, Deno may be a better a choice as https://npmtrends.com/ show than Bun usage is trending down (see 5 year graph) and they've yet to ship a stable `1.x`.

Maybe this issue should be closed and instead replaced with an epic to migrate to Deno (if that's what you're happy to migrate to)?

**Comment by dae** [comment]
Interesting to hear that Bun is trending down - I guess the recent concessions deno has made like supporting npm packages have helped.

I've created a new ticket regarding a switch to deno: #2439
